### ISSUE TITLE
Possible bug in VariableDatum

### ISSUE THREAD

**Issue by IanMayo** [open-dis/open-dis-java#4]
I'm not sure which is the current master copy of java sources, but I have a problem with VariableDatum.java, as recorded in the SVN head [here](https://sourceforge.net/p/open-dis/code/HEAD/tree/languages/java/trunk/src/main/java/edu/nps/moves/dis/VariableDatum.java#l95)

````
public void setVariableData(byte[] pVariableData)
{ 
    int unitsOfEight = pVariableData.length / 8;
    int remainder = pVariableData.length % 8;
    
    if(remainder != 0)
        unitsOfEight++;
    
    byte[] newData = new byte[unitsOfEight];
    for(int idx = 0; idx < pVariableData.length; idx++)
    {
        newData[idx] = pVariableData[idx];
    }
    
    variableData = newData;
    variableData = pVariableData;
}
````

I have two issues.  The first is an array index problem in the ````newData[idx] = ```` call.  newData is an array of (roughly) pVariableData.length / 8.  But, we're assigning to it using an array loop of pVariableData.length.  This causes the array index problem.

The second is that we assign the new data array to variableData, then we assign the raw pVariableData array to variableData, negating our 8-byte array transformation performed earlier in the method.

I *guess* the intended behaviour is that newData should be an array of 8-byte arrays.  If this is correct, then I'm happy to create a PR accordingly.  But, my guess could be way off.

**Comment by IanMayo** [comment]
It looks like this issue has been resolved in 970fbd40b8a22060dc03157782d6b7ba1afad604

**Comment by IanMayo** [comment]
Resolved in 970fbd40b8a22060dc03157782d6b7ba1afad604
